static/elements/chromedash-upcoming.js

```diff
@@ -52,12 +52,110 @@ class ChromedashUpcoming extends LitElement {
       cardWidth: {type: Number},
+      lastFetchedOn: {type: Number},
+      lastMilestoneVisible: {type: Number},
+      milestoneArray: {type: Array},
+      milestoneInfo: {type: Object},
+      cardsToFetchInAdvance: {type: Number},
     };
   }
 
+  set lastFetchedOn(val) {
+    let oldVal = this._lastFetchedOn;
+    this._lastFetchedOn = val;
+    this.fetchNextBatch().then(()=>{
+      this.requestUpdate('lastFetchedOn', oldVal);
+    }).catch(() => {
+      alert('Some error occurred. Please refresh the page or try again later.');
+    });
+  }
+
+  get lastFetchedOn() {
+    return this._lastFetchedOn;
+  }
+
   constructor() {
     super();
     this.cardWidth = this.computeWidthofCard();
+    this.starredFeatures = new Set();
+    this.milestoneArray = [];
+    this.milestoneInfo = {};
+    this.cardsToFetchInAdvance = 3;
+  }
+
+  async fetchNextBatch() {
+    // promise to fetch next milestones
+    const nextMilestones = window.csClient.getSpecifiedChannels(this._lastFetchedOn+1+1,
+      this._lastFetchedOn+1+this.cardsToFetchInAdvance);
+
+    let tempMilestoneArray = [];
+    for (let i = 1; i <= this.cardsToFetchInAdvance; i++) {
+      tempMilestoneArray.push(this._lastFetchedOn+1+i);
+    }
+
+    // promise to fetch features in next milestones
+    let milestoneFeaturePromise = {};
+    tempMilestoneArray.forEach((milestone) => {
+      milestoneFeaturePromise[milestone] = window.csClient.getFeaturesInMilestone(milestone);
+    });
+
+    let tempMilestoneInfo;
+    let milestoneFeatures = {};
+
+    try {
+      tempMilestoneInfo = await nextMilestones;
+
+      for (let milestone of tempMilestoneArray) {
+        milestoneFeatures[milestone] = await milestoneFeaturePromise[milestone];
+      }
+    } catch (err) {
+      throw (new Error('Unable to load Features'));
+    }
+
+
+    // add some details to milestone information fetched
+    tempMilestoneArray.forEach((milestone) => {
+      tempMilestoneInfo[milestone].version = milestone;
+      tempMilestoneInfo[milestone].features =
+          this.mapFeaturesToShippingType(milestoneFeatures[milestone]);
+    });
+
+    // update the properties to render the latest milestone cards
+    this.milestoneInfo = Object.assign({}, this.milestoneInfo, tempMilestoneInfo);
+
+    this.milestoneArray = [...this.milestoneArray, ...tempMilestoneArray];
+  }
+
+  mapFeaturesToShippingType(features) {
+    const featuresMappedToShippingType = {};
+    features.forEach(f => {
+      const component = f.browsers.chrome.status.text;
+      if (!featuresMappedToShippingType[component]) {
+        featuresMappedToShippingType[component] = [];
+      }
+      featuresMappedToShippingType[component].push(f);
+    });
+
+    for (let [, feautreList] of Object.entries(featuresMappedToShippingType)) {
+      this.sortFeaturesByName(feautreList);
+    }
+
+    return featuresMappedToShippingType;
+  }
+
+
+  sortFeaturesByName(features) {
+    features.sort((a, b) => {
```

Comment:
Try 
features.sort((a, b) =>
    a.localeCompare(b, 'en', {'sensitivity': 'base'}) 
);
which is based off a comment in https://stackoverflow.com/questions/8996963/how-to-perform-case-insensitive-sorting-in-javascript

I don't expect non-English feature names, but that code is shorter.

Reply:
Okay. I have made this change. 
For the record, the last two arguments `('en', {'sensitivity': 'base'})` are not supported by Webview Android and Opera for Android.

Reply:
I didn't know about the lack of support.  In that case, how about this?
```
features.sort((a, b) =>
    a.name.toLowerCase().localeCompare(b.name.toLowerCase())
);
```

Reply:
Done. See [this](https://github.com/GoogleChrome/chromium-dashboard/pull/1458/commits/9881f927610b11f5577ad1a8183fa7918686b639) commit